### ISSUE TITLE
Not working with default config

### ISSUE THREAD

**Issue by sangyye** [bachya/pinpress#2]
I installed the gem with ruby 2.2.0 on MacOSX. I run `pinpress init` and get already this error message `/usr/local/lib/ruby/gems/2.2.0/gems/htmlentities-4.3.1/lib/htmlentities/mappings/expanded.rb:465: warning: duplicated key at line 466 ignored: "inodot"`, although the gem runs okay.
So i could run the init command without further problems and get the default config. I tried to run `pinpress pins` to test it but i only get:

```
/usr/local/lib/ruby/gems/2.2.0/gems/htmlentities-4.3.1/lib/htmlentities/mappings/expanded.rb:465: warning: duplicated key at line 466 ignored: "inodot"
# Invalid template specified and/or no default template defined
```

So the config look like this:

```

---
pinpress:
  config_location: "/Users/sangyye/.pinpress"
  default_pin_template: pinpress_default
  default_tag_template: pinpress_default
  log_level: WARN
  version: 1.6.1
  api_token: MYAPIKEY
pin_templates:
- name: pinpress_default
  opener: "<ul>"
  item: "<li><%= href %></li>"
  closer: "</ul>"
tag_templates:
- name: pinpress_default
  item: "<%= tag %> (<%= count %>),"
```

So it looks total different form the examples in the readme, so I thought I change it to the format in the readme. But I only get: `# (/Users/sangyye/.pinpress): mapping values are not allowed in this context`. So I can't get this running, do you have an idea?

**Comment by bachya** [comment]
Hi there,

For your first question, take a look at [this Stackoverflow questions/answer](https://stackoverflow.com/questions/27669855/warnings-after-upgrading-to-ruby-2-2-0) – long story short, there's always a danger in running the latest and greatest Ruby before gems have been properly updated. `htmlentities` looks like the culprit here.

For the second question, your template layouts are incorrect. If we look at just your one PinPress template, try changing this:

``` yaml
pin_templates:
- name: pinpress_default
  opener: "<ul>"
  item: "<li><%= href %></li>"
  closer: "</ul>"
```

...to...

``` yaml
pin_templates:
- pinpress_default:
    opener: "<ul>"
    item: "<li><%= href %></li>"
    closer: "</ul>"
```

**Comment by sangyye** [comment]
Hi,

I just use the ruby version that is offered by homebrew, and it's the latest stable from ruby.

I change the config like you said, still 

```
/usr/local/lib/ruby/gems/2.2.0/gems/htmlentities-4.3.1/lib/htmlentities/mappings/expanded.rb:465: warning: duplicated key at line 466 ignored: "inodot"
# (/Users/sangyye/.pinpress): mapping values are not allowed in this context
# at line 11 column 11
```

**Comment by bachya** [comment]
Hi @sangyye,

Run `gem update pinpress` – I just released version 1.6.2, which includes the latest htmlentities gem. After upgrading, when I run `pinpress pins`, I see the data I expect. Let me know?

**Comment by sangyye** [comment]
Hi @bachya,

thanks for answering me. I directly make an update and run `pinpress pins` but i only get an error with
 `# undefined method `load_file' for Psych:Module`

Update:
Ok, after removing the old packages I get this error, i tried all 3 possible config options, the error stays the same:

```
# (/Users/sangyye/.pinpress): mapping values are not allowed in this context
# at line 11 column 9
```

**Comment by bachya** [comment]
Hi @sangyye,

Could you paste your entire `~/.pinpress`? Is it still what I've pasted below?

``` yaml
---
pinpress:
  config_location: "/Users/sangyye/.pinpress"
  default_pin_template: pinpress_default
  default_tag_template: pinpress_default
  log_level: WARN
  version: 1.6.1
  api_token: MYAPIKEY
pin_templates:
- name: pinpress_default
  opener: "<ul>"
  item: "<li><%= href %></li>"
  closer: "</ul>"
tag_templates:
- name: pinpress_default
  item: "<%= tag %> (<%= count %>),"
```

**Comment by sangyye** [comment]
Hi @bachya,

Of course, it looks like this:

``` yaml
---
pinpress:
  config_location: "/Users/sangyye/.pinpress"
  default_pin_template: pinpress_default
  default_tag_template: pinpress_default
  log_level: WARN
  version: 1.6.1
  api_token: MYAPIKEY
pin_templates:
- name: pinpress_default
  opener: "<ul>"
  item: "<li><%= href %>"
  closer: "</ul>"
- name: markdown
  item: "* <%= href %>"
tag_templates:
- name: pinpress_default
  item: "<%= tag %> (<%= count %>),"
```

Found a little error in my config. Now I get this if i call `pinpress pins`:

```
# Invalid template specified and/or no default template defined
```

I tried `pinpress templates` and get this:

```
---> AVAILABLE PIN TEMPLATES:
1.  Name:   name
# no implicit conversion of Symbol into Integer
```

Thanks for your help

**Comment by bachya** [comment]
Hi @sangyye,

It looks like you still have the incorrect format for your templates; see my first comment for more info.

If you change to this:

``` yaml
---
pinpress:
  config_location: "/Users/sangyye/.pinpress"
  default_pin_template: pinpress_default
  default_tag_template: pinpress_default
  log_level: WARN
  version: 1.6.1
  api_token: MYAPIKEY
pin_templates:
- pinpress_default
    opener: "<ul>"
    item: "<li><%= href %>"
    closer: "</ul>"
- markdown
    item: "* <%= href %>"
tag_templates:
- pinpress_default
    item: "<%= tag %> (<%= count %>),"
```

...I bet it'll work. Let me know?

**Comment by sangyye** [comment]
Hi @bachya,

I change the config like above and I only get:

```
# (/Users/sangyye/.pinpress): mapping values are not allowed in this context
# at line 11 column 11
```

So it should be this line `opener: "<ul>"`

**Comment by bachya** [comment]
I ignored one important item: you need to include a `:` after the template name. So, your config file would look like this:

``` yaml
---
pinpress:
  config_location: "/Users/sangyye/.pinpress"
  default_pin_template: pinpress_default
  default_tag_template: pinpress_default
  log_level: WARN
  version: 1.6.1
  api_token: MYAPIKEY
pin_templates:
- pinpress_default:
    opener: "<ul>"
    item: "<li><%= href %>"
    closer: "</ul>"
- markdown:
    item: "* <%= href %>"
tag_templates:
- pinpress_default:
    item: "<%= tag %> (<%= count %>),"
```

**Comment by sangyye** [comment]
Hey @bachya,
thanks a lot, it's now working. Many thanks to you.
Two little things, `pinpress init` generate a wrong config file in default. So it's not working directly.
The other is in the Readme the config Schema is also not correct.
Feel free to close this issue my problem is solve now, thanks a lot :+1:

**Comment by bachya** [comment]
Hi @sangyye,

Glad to hear it's working!

Thanks so much for letting me know about those two small issues. I'm going to leave this ticket open until I correct; will close once they're resolved.

**Comment by sangyye** [comment]
Hi @bachya,

Thanks for helping me out. Glad I could help you :smile:

**Comment by bachya** [comment]
All fixed! Version 1.6.3 should do the trick. Thanks!
